### ISSUE TITLE
Getting authentication Error upon Pairing

### ISSUE THREAD

**Issue by rmervine** [alex20465/deskbluez#21]
I'm having an issue with getting deskbluez to pair to my desk.  What happens is that it sees the desk, but then sees the authentication fails. However, the desk does say "paired" on the desk and it's also reflected in bluetoothctl. I can confirm nothing else is pairing during this process. What I done is that when the desk is in pair mode, I waited 30 seconds to pair just to make sure there is nothing conflicting. And I removed the device out of bluetoothctl and even confirmed that nothing was pairing when running the command. Also, I can pair with no problems through my IOS app and the Win10 app.

`[bluetooth]# paired-devices`
`[bluetooth]# exit`
`pi@raspberrypi:~ $ sudo deskbluez --debug  connect`
`[DEBUG] CONFIG: load { configPath: '/root/.config/deskbluez-default',
  profile: 'default' }`
`[DEBUG] BLUETOOTH: init`
`[DEBUG] BLUETOOTH: use adapter hci0`
`[DEBUG] BLUETOOTH: adapter loaded { Address: 'B8:27:EB:17:DA:01',`
 ` AddressType: 'public',`
  `Name: 'raspberrypi',`
  `Alias: 'raspberrypi',`
  `Class: 0,`
  `Powered: true,`
  `Discoverable: false,`
 ` DiscoverableTimeout: 180,`
  `Pairable: true,`
  `PairableTimeout: 0,`
  `Discovering: false,`
 ` UUIDs:`
  `[ '00001801-0000-1000-8000-00805f9b34fb',`
     `'0000110e-0000-1000-8000-00805f9b34fb',`
     `'00001200-0000-1000-8000-00805f9b34fb',`
     `'0000110c-0000-1000-8000-00805f9b34fb',`
    ` '00001800-0000-1000-8000-00805f9b34fb' ],`
  `Modalias: 'usb:v1D6Bp0246d0532' }`
`✔ make sure the desk device isn't paired OR connected … yes
`✔ choose device model: › linak
`✔ enable the pairing mode (usually hold bluetooth button for 2-3 seconds) … yes
`[DEBUG] BLUETOOTH: start discovery
`[DEBUG] BLUETOOTH: device discovered CA:45:74:C1:A7:7E { Address: 'CA:45:74:C1:A7:7E',
 ` AddressType: 'random',
 ` Name: 'DESK 2878',`
 ` Alias: 'DESK 2878',`
 ` Paired: false,`
 ` Trusted: false,`
 ` Blocked: false,`
 ` LegacyPairing: false,`
 ` RSSI: -51,`
  `Connected: false,`
  `UUIDs: [ '99fa0001-338a-1024-8a49-009c0215f78a' ],`
  `Adapter: '/org/bluez/hci0',`
  `TxPower: 4,`
  `ServicesResolved: false }`
`[DEBUG] BLUETOOTH: stop discovery`
`✔ choose desk device › DESK 2878 [CA:45:74:C1:A7:7E]`
`[DEBUG] BLUETOOTH: connect to CA:45:74:C1:A7:7E`
`[WARN] { Error: Authentication Failed`
 `   at new DBusError (/usr/local/lib/node_modules/deskbluez/node_modules/dbus/lib/error.js:9:9)`
 `   at createError (/usr/local/lib/node_modules/deskbluez/node_modules/dbus/lib/bus.js:243:9)`
 ` message: 'Authentication Failed',`
 ` dbusName: 'org.bluez.Error.AuthenticationFailed' }`
`Authentication Failed`

**Comment by rmervine** [comment]
Also I tried the report functionality:

pi@raspberrypi:~ $ sudo deskbluez report
✔ make sure the desk device isn't paired OR connected … yes
✔ enable the pairing mode (usually hold bluetooth button for 2-3 seconds) … yes
✔ choose desk device › DESK 2878 [CA:45:74:C1:A7:7E]
Authentication Failed
pi@raspberrypi:~ $

**Comment by rmervine** [comment]
I have made some additional interesting findings. I tried looking at the behavior in bluetoothctl. If I tried to pair the Linak directly in there, I would see the same "authentication failed" message. Now if I trusted the device in bluetoothctl, I could then pair successfully as bluetoothctl would ask if I wanted to pair and I choose "yes" Now if the device is paired in bluetoothctl, it doesn't show in deskbluez. And that may be by design. If I remove the device in bluetoothctl, then deskbluez sees the device. But again I get the failed authentication message.

**Comment by alex20465** [comment]
If you see the error is raised by `bluez` (`org.bluez.Error.AuthenticationFailed`).

Pairing the device through bluetoothctl will not work, it needs to be unpaired and forgotten to get it work. 

-  try to disable the scan and enable the agent via bluetoothctl
-  make sure the bluetooth service is enabled and running (systemctl).
- try to use deskbluez with a non-root user (without sudo)

**Comment by rmervine** [comment]
@alex20465 

Thank you for the quick response. I do realize that the device can't be paired with bluetoothctl first or the device won't even show up during discovery. I was just showing something I noticed when troubleshooting

I went in bluetoothctl and confirmed the device is not paired And I also turned scanning off. I then confirmed via systemctl that bluetooth is enabled

`pi@raspberrypi:~ $ sudo bluetoothctl`
`Agent registered`
`[bluetooth]# list`
`Controller B8:27:EB:17:DA:01 raspberrypi [default]`
`[bluetooth]# paired-devices`
`[bluetooth]# scan off`
`Failed to stop discovery: org.bluez.Error.Failed`
`[bluetooth]# exit`

`pi@raspberrypi:~ $ sudo systemctl status bluetooth.service`
`● bluetooth.service - Bluetooth service`
 `  Loaded: loaded (/lib/systemd/system/bluetooth.service; enabled; vendor preset`
  ` Active: active (running) since Sun 2021-09-05 21:54:14 BST; 17h ago`
   `  Docs: man:bluetoothd(8)`
 `Main PID: 477 (bluetoothd)`
  ` Status: "Running"`
   ` Tasks: 1 (limit: 2062)`
   `CGroup: /system.slice/bluetooth.service`
    `       └─477 /usr/lib/bluetooth/bluetoothd`

Now I did see this below the status
`Sep 06 02:12:42 raspberrypi bluetoothd[477]: No agent available for request type 2`
`Sep 06 02:12:42 raspberrypi bluetoothd[477]: device_confirm_passkey: Operation not permitted`

I retried the pairing and getting the same issue.

I went back in to bluetoothctl and I see my desk name in there, but not paired:

`pi@raspberrypi:~ $ sudo bluetoothctl`
`Agent registered`
`[DESK 2878]# paired-devices`
`[DESK 2878]# `

And when I looked at systemctl, I confirmed the same errors I posed above happened when I tried to pair:

 `Sep 06 15:34:00 raspberrypi bluetoothd[477]: No agent available for request type 2`
`Sep 06 15:34:00 raspberrypi bluetoothd[477]: device_confirm_passkey: Operation not permitted`
`Sep 06 15:34:02 raspberrypi bluetoothd[477]: No agent available for request type 2`
`Sep 06 15:34:02 raspberrypi bluetoothd[477]: device_confirm_passkey: Operation not permitted`

And if I run this without sudo, I get these errors and the program never runs

`pi@raspberrypi:~ $ deskbluez  --debug connect`
`[DEBUG] CONFIG: load { configPath: '/home/pi/.config/deskbluez-default',`
 ` profile: 'default' }`
`[DEBUG] BLUETOOTH: init`
`(node:3309) UnhandledPromiseRejectionWarning: Error: No introspectable`
 `   at /usr/local/lib/node_modules/deskbluez/node_modules/dbus/lib/bus.js:131:15`
`(node:3309) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing` `inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection` `id: 1)`
`(node:3309) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise` `rejections that are not handled will terminate the Node.js process with a non-zero exit code.`

It almost sounds like permissions issue somewhere even when running as root. Any ideas?
